src/samples.jl

```diff
@@ -423,6 +423,30 @@ function load(file_path, file_format::AbstractLPCMFormat, info::SamplesInfo, spa
     return encoded ? samples : decode(samples)
 end
 
+"""
+    Onda.mmap(signal)
+
+Return `Onda.mmap(signal.file_path, SamplesInfo(signal))`, throwing an `ArgumentError` if `signal.file_format != "lpcm"`.
+"""
+function mmap(signal)
```

Comment:
IMO it would make more sense to extend `Mmap.mmap`, as that's what it's doing. This already assumes that the object passed has `file_format` and `file_path` fields and a `SamplesInfo` constructor method, so in reality the duck typing won't buy you much over just making it `Mmap.mmap(::Signal)`.

Reply:
`SamplesInfo` constructor is also duck-typed.

Would be quite weird/inconsistent with the rest of the API to introduce a type barrier here...would be annoying to force users to wrap e.g. transformed table results in `Signal` only for this function. 

AFAICT callers won't be treating `Mmap.mmap` generically over its Base definition's input domain AND this input domain at the same time, so not sure I see any benefit to the extending the existing method

Reply:
You're already leaning pretty hard on this behaving like a `Signal` -- and moreover a particular variant of that -- so why not add that barrier?  If users want to use a wonky custom type, then they can write their own method -- they can see the source here and adapt it for their own type.

I would also be irritated by the name collision caused by not extending because if I'm memory mapping with one thing, I'm probably doing it in a lot things and thus importing `Mmap`

Reply:
> If users want to use a wonky custom type, then they can write their own method -- they can see the source here and adapt it for their own type.

Eh, that's unnecessary/not the alternative at play here. The choice here is essentially `Mmap.mmap(Signal(x))` vs. `Onda.mmap(x)`

> I would also be irritated by the name collision caused by not extending because if I'm memory mapping with one thing, I'm probably doing it in a lot things and thus importing Mmap

There's no name collision as neither `Onda` nor `Mmap` export `mmap`...to cause a name collision you'd purposefully have to import both

Choosing to duck-type this sort of functionality was already an intentional decision made as part of the v0.12 update (which I stand by - it's VERY convenient when working within the Tables.jl ecosystem). To put it another way: even if we all decided to walk back that duck-type choice, we should do so in a way that's consistent across the API, not introduce an inconsistency. 

So if we want to bikeshed the duck-typed nature of the interface let's do it in a separate issue. If we did decide to change this it'd be an easy enough deprecation path anyway.

Reply:
>Eh, that's unnecessary/not the alternative at play here. The choice here is essentially Mmap.mmap(Signal(x)) vs. Onda.mmap(x)

That actually seems like a strong argument to type-restrict `Onda.mmap`....

Reply:
> So if we want to bikeshed the duck-typed nature of the interface let's do it in a separate issue. If we did decide to change this it'd be an easy enough deprecation path anyway.

given the above i'm going to mark this as resolved - i don't think this particular bikeshed should block this otherwise relatively small/quick PR

> That actually seems like a strong argument to type-restrict Onda.mmap....

i'm not sure i see what you're getting at but feel free to elaborate in an issue

Reply:
I had meant to re-review this but got bogged down with other things. If it needs to be duck typed then IMO a better interface would be `mmap_samples` and export it.